### PR TITLE
Update documentation init_process_group optional backend

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94543

Update documentation for `init_process_group()` to mention the `backend` argument is optional.

### CODE REVIEW

File: test/distributed/test_c10d_common.py [hunk 1]
```diff
@@ -1532,6 +1532,18 @@ def tearDown(self):
...
+            store = dist.FileStore(f.name, self.world_size)
+            # creates both gloo and nccl backend
+            if dist.is_gloo_available() and dist.is_nccl_available():
+                dist.init_process_group(
+                    store=store,
+                    rank=self.rank,
+                    world_size=self.world_size,
+                )
+                dist.destroy_process_group()
+
```
Comment (kwen2501) [inline test/distributed/test_c10d_common.py:1546]: Would you like to add two all_reduce calls (one with cpu, the other with gpu) to make sure it is actually working? :)
If we already have such test, please ignore me :)
Also, it can be in a separate PR if we don't have it yet.

**Review by kwen2501 — APPROVED** [review]
Looks good! Thanks!

File: test/distributed/test_c10d_common.py (hunk 1 above)
Comment (H-Huang) [reply test/distributed/test_c10d_common.py:1546]: Sounds good, it should work since we support gloo + nccl subgroups which would follow similar logic as the new backends. I will definitely refactor the existing tests for dispatchable collectives to include this though
